Source/SIMPLib/DataContainers/DataContainerProxy.cpp

```diff
@@ -228,10 +228,81 @@ void DataContainerProxy::setFlags(uint8_t flag, AttributeMatrixProxy::AMTypeFlag
   for(QMap<QString, AttributeMatrixProxy>::iterator amIter = attributeMatricies.begin(); amIter != attributeMatricies.end(); ++amIter) // AttributeMatrix Level
   {
     AttributeMatrixProxy& amProxy = amIter.value();
-    AttributeMatrixProxy::AMTypeFlag amTypeFlag = AttributeMatrixProxy::AttributeMatrixTypeToFlag(static_cast<AttributeMatrix::Type>(amProxy.amType));
+    AttributeMatrixProxy::AMTypeFlag amTypeFlag = AttributeMatrixProxy::AttributeMatrixTypeToFlag(static_cast<AttributeMatrix::Type>(amProxy.getAMType()));
     if ((amTypes & amTypeFlag) > 0 || amTypes == AttributeMatrixProxy::AMTypeFlag::Any_AMType)
     {
       amProxy.setFlags(flag, primitiveTypes, compDimsVector);
     }
   }
 }
+
+QMap<QString, AttributeMatrixProxy> DataContainerProxy::getAttributeMatricies()
+{
+	return attributeMatricies;
+}
+
+void DataContainerProxy::setAttributeMatricies(const QMap<QString, AttributeMatrixProxy>& newAttributeMatricies)
+{
+	attributeMatricies = newAttributeMatricies;
+}
+
+QString DataContainerProxy::getName() const
+{
+	return name;
+}
+
+void DataContainerProxy::setName(const QString& newName)
+{
+	name = newName;
+}
+
+uint8_t DataContainerProxy::getFlag() const
+{
+	return flag;
+}
+
+void DataContainerProxy::setFlag(uint8_t newFlag)
+{
+	flag = newFlag;
+}
+
+void DataContainerProxy::setDCType(uint32_t dType)
+{
+  dcType = dType;
+}
+
+uint32_t DataContainerProxy::getDCType() const
+{
+  return dcType;
+}
+
+void DataContainerProxy::toggleFlag()
+{
+	if (flag == Qt::Checked)
+	{
+		flag = Qt::Unchecked;
+	}
+	else
+	{
+		flag = Qt::Checked;
+	}
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+AttributeMatrixProxy& DataContainerProxy::getAttributeMatrixProxy(const QString& name)
+{
+	for (QMap<QString, AttributeMatrixProxy>::iterator amIter = attributeMatricies.begin(); amIter != attributeMatricies.end(); ++amIter) // AttributeMatrix Level
+	{
+		AttributeMatrixProxy& am = amIter.value();
+    if(am.getName() == name)
```

Comment:
An if statement inside a for loop should not be less indented than the for that contains it.

Reply:
Also, `AttributeMatrixProxy proxy(name);` is oddly indented a few lines after this to match the for loop.